api/utils/accessio/chunkedreader.go

```diff
@@ -0,0 +1,118 @@
+package accessio
+
+import (
+	"bytes"
+	"io"
+	"sync"
+
+	"github.com/mandelsoft/goutils/general"
+)
+
+// ChunkedReader splits a reader into several
+// logical readers with a limited content size.
+// Once the reader reaches its limits it provides
+// a io.EOF.
+// It can be continued by Calling Next, which returns
+// whether a follow-up is required or not.
+type ChunkedReader struct {
+	lock   sync.Mutex
+	reader io.Reader
+	buffer *bytes.Buffer
+	size   int64
+	chunk  int
+	read   int64
+	err    error
+
+	preread uint
+}
+
+var _ io.Reader = (*ChunkedReader)(nil)
+
+func NewChunkedReader(r io.Reader, chunk int64, preread ...uint) *ChunkedReader {
```

Comment:
and here, chunk is the size of the chunks?

Reply:
yes, changed